### DISCUSSION TITLE
Select doesn't re-select right option when the order changes

### DISCUSSION THREAD

**Discussion by danielbehrendt** [livewire/livewire#7957] [7. Bugs (archived)]
### Livewire version

v3.4.4

### Laravel version

v10.44.0

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

When having an array of options for a select field which is live and the order isn't consistent (could be shuffled or an item could be added/removed) between requests most of the times the wrong option gets selected. The option which is selected after a request has the same index as before but it's probably not the same "value" as before.

See the simple setup of the Wirebox and change the select field multiple times to see that it's jumping.

### Repository or Wirebox URL

https://wirebox.app/b/y821d

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment]
the classic... and it's not bug (see https://github.com/livewire/livewire/pull/7500)

please add `wire:key` to the select input...

```blade
    <select wire:model.live="ingredient" wire:key="ingredient-{{ $ingredient }}">
        @foreach ($this->getIngredients() as $value => $name)
            <option value="{{ $value }}">{{ $name }}</option>
        @endforeach
    </select>

    <span>{{ $ingredient }}</span>
```
